docs/api/app.md

```diff
@@ -742,33 +758,31 @@ app.setJumpList([
 ])
 ```
 
-### `app.makeSingleInstance(callback)`
-
-* `callback` Function
-  * `argv` String[] - An array of the second instance's command line arguments
-  * `workingDirectory` String - The second instance's working directory
-
-Returns `Boolean`.
+### `app.makeSingleInstance()`
 
 This method makes your application a Single Instance Application - instead of
 allowing multiple instances of your app to run, this will ensure that only a
 single instance of your app is running, and other instances signal this
 instance and exit.
 
-`callback` will be called by the first instance with `callback(argv, workingDirectory)`
-when a second instance has been executed. `argv` is an Array of the second instance's
-command line arguments, and `workingDirectory` is its current working directory. Usually
-applications respond to this by making their primary window focused and
-non-minimized.
+### `app.isSingleInstance()`
 
-The `callback` is guaranteed to be executed after the `ready` event of `app`
-gets emitted.
+Returns `Boolean`
+
+This methods returns whether or not `makeSingleInstance` has been called.
 
-This method returns `false` if your process is the primary instance of the
-application and your app should continue loading. And returns `true` if your
+### `app.isPrimaryInstance()`
+
+Returns `Boolean`
+
+This method returns `true` if your process is the primary instance of the
+application and your app should continue loading. And returns `false` if your
 process has sent its parameters to another instance, and you should immediately
 quit.
 
+This method will throw an error if you call it without first calling
+`app.makeSingleInstance()`.
```

Comment:
Because `makeSingleInstance` returning a boolean that indicates whether it is the second instance doesn't semantically make a whole lot of sense IMO.  This might be API preference creeping in but I 100% prefer a slightly more verbose self-explanatory API than an API that doesn't make sense without reading the docs to figure out what the return value of `makeSingleInstance` means.

Like I've seen this before:

```js
if (app.makeSingleInstance(cb)) {
  return app.quit();
}
```

Which although is API correct, makes no sense to anyone reading that.  Especially when most "action" api's that Electron exposes the boolean value indicates success.

Reply:
We're in agreement the current API is unintuitive -- the proposed API is more readable and has return values that make sense.

However, I'm not sure how useful these new methods are. I don't see a use case for  `IsSingleInstance()` and it doesn't seem that `makeSingleInstance()` and `IsPrimaryInstance()` make use except when called as a pair, suggesting that they could be folded into a single function.

Which comes back to the current API: putting aside the unintuitive return value and slightly opaque name for the moment, it does have strengths in that it's impossible to call the pieces out of order, to miss a call, or to forget to register for a signal. It's very difficult to get wrong.

So what if we preserved that while fixing the name and return value? Something like:

```js
if (!app.registerSecondInstanceHandler(cb)) {
  app.quit();
}
```
This seems pretty readable to me.